services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.repository.ts

```diff
@@ -16,47 +20,119 @@ export class ProgramFinancialServiceProviderConfigurationRepository extends Repo
     );
   }
 
-  public async findByProgramIdAndFinancialServiceProviderName(
-    programId: number,
-    financialServiceProviderName: FinancialServiceProviderName,
-  ): Promise<ProgramFinancialServiceProviderConfigurationEntity[]> {
+  public async findByProgramIdAndFinancialServiceProviderName({
+    programId,
+    financialServiceProviderName,
+    includeProperties,
+  }: {
+    programId: number;
+    financialServiceProviderName: FinancialServiceProviders;
+    includeProperties: boolean;
+  }): Promise<ProgramFinancialServiceProviderConfigurationEntity[]> {
     return await this.baseRepository.find({
       where: {
         programId: Equal(programId),
-        fsp: { fsp: Equal(financialServiceProviderName) },
+        financialServiceProviderName: Equal(financialServiceProviderName),
       },
+      relations: includeProperties ? ['properties'] : [],
     });
   }
 
-  public async getValuesByNamesOrThrow({
-    programId,
-    financialServiceProviderName,
+  public async findUsernamePasswordPropertiesForIds(
+    programFinancialServiceProviderConfigurationId: number[],
+  ): Promise<
+    {
+      programFinancialServiceProviderConfigurationId: number;
+      credentials: UsernamePasswordInterface;
+    }[]
+  > {
+    return await Promise.all(
+      programFinancialServiceProviderConfigurationId.map(async (id) => ({
+        programFinancialServiceProviderConfigurationId: id,
+        credentials: await this.getUsernamePasswordProperties(id),
+      })),
+    );
+  }
+
+  public async getUsernamePasswordProperties(
+    programFinancialServiceProviderConfigurationId: number,
```

Comment:
I think it may be clearer to leave it like this. Because people may be confused if it is programFinancialServiceProviderConfigurationId or programFinancialServiceProviderConfigurationPropertyId.